pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -7,12 +7,43 @@
 interface=$1
 action=$2
 
+main() {
+    if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]] || [[ $interface == enP* && $action == "up" ]] || [[ $interface == enP* && $action == "down" ]]; then
```

Comment:
Do we have a manual test plan for this PR / way to simulate this behavior before we release it? I think confidence is key since it takes a lot of effort and communication to get these changes out to customers.

Although the host servicing events are random, maybe we can remove the resolv.conf in a test cluster to confirm expected logging, and restart the NIC with network manager to confirm that dnsmasq service restarts without issues in all supported ARO OCP versions (should only be 2-3 versions).